tests/upgrades/test_upgrades.py

```diff
@@ -17,334 +19,351 @@
-    )
-    proxy = await starknet.deploy(
-        contract_class=proxy_cls,
-        constructor_calldata=[v1_decl.class_hash]
-    )
-    return (
-        starknet.state,
-        account1,
-        account2,
-        v1_decl,
-        v2_decl,
-        proxy
-    )
-
-
-@pytest.fixture
-def proxy_factory(contract_classes, proxy_init):
-    account_cls, _, _, proxy_cls = contract_classes
-    state, account1, account2, v1_decl, v2_decl, proxy = proxy_init
-    _state = state.copy()
-    account1 = cached_contract(_state, account_cls, account1)
-    account2 = cached_contract(_state, account_cls, account2)
-    proxy = cached_contract(_state, proxy_cls, proxy)
-
-    return account1, account2, proxy, v1_decl, v2_decl
-
-
-@pytest.fixture
-async def after_upgrade(proxy_factory):
-    admin, other, proxy, v1_decl, v2_decl = proxy_factory
-
-    # initialize, set value, and upgrade to v2
-    await signer.send_transactions(
-        admin,
-        [
-            (proxy.contract_address, 'initializer', [admin.contract_address]),
-            (proxy.contract_address, 'setValue1', [VALUE_1]),
-            (proxy.contract_address, 'upgrade', [v2_decl.class_hash])
-        ]
-    )
-
-    return admin, other, proxy, v1_decl, v2_decl
-
-
-@pytest.mark.asyncio
-async def test_initializer(proxy_factory):
-    admin, _, proxy, *_ = proxy_factory
-
-    await signer.send_transaction(
-        admin, proxy.contract_address, 'initializer', [
-            admin.contract_address
-        ]
-    )
+class TestUpgrades:
+    @pytest.fixture(scope='module')
+    def contract_classes(self):
+        account_cls = Account.get_class
+        v1_cls = get_contract_class('UpgradesMockV1')
+        v2_cls = get_contract_class('UpgradesMockV2')
+        proxy_cls = get_contract_class('Proxy')
+
+        return account_cls, v1_cls, v2_cls, proxy_cls
+
+
+    @pytest.fixture(scope='module')
+    async def implementations_declare(self, contract_classes):
+        _, v1_cls, v2_cls, proxy_cls = contract_classes
+        starknet = await State.init()
+        account1 = await Account.deploy(signer.public_key)
+        account2 = await Account.deploy(signer.public_key)
+
+        v1_decl = await starknet.declare(
+            contract_class=v1_cls,
+        )
+        v2_decl = await starknet.declare(
+            contract_class=v2_cls,
+        )
+
+        return (
+            starknet,
+            account1,
+            account2,
+            v1_decl,
+            v2_decl,
+            proxy_cls
+        )
+
+
+    @pytest.fixture(scope='module')
+    async def proxy_deploy(self, implementations_declare):
+        starknet, account1, account2, v1_decl, v2_decl, proxy_cls = implementations_declare
+
+        # with selector set to 0, internal initialization call must be ignored
+        proxy = await starknet.deploy(
+            contract_class=proxy_cls,
+            constructor_calldata=[
+                v1_decl.class_hash,
+                0,
+                0, 
```

Comment:
```suggestion
                0,
```
Sorry 😅 force of habit

Reply:
Don't be sorry, I don't know how I keep missing them, do you have an extension for it?

Reply:
Yes, sir:
https://marketplace.visualstudio.com/items?itemName=shardulm94.trailing-spaces